lib/src/coap_client.dart

```diff
@@ -364,7 +379,9 @@ class CoapClient {
         final socket = CoapINetwork.fromUri(uri,
             address: destination,
             config: _config,
-            namespace: _eventBus.namespace);
+            namespace: _eventBus.namespace,
+            ecdsaKeys: ecdsaKeys,
+            pskCredentialsCallback: request.pskCredentialsCallback);
```

Comment:
Since the credentials callback can be set on every request, and the endpoint is re-used (and potentially used concurrently by multiple requests), perhaps we need to think of another way of passing the `pskCredentialsCallback`.

Do we expect it to change from one request to another or could it be set at another level, for example as an optional parameter passed to the client itself?

Reply:
Setting it at the client level would certainly be an option, especially since connections are reused between requests.

Reply:
Moved the callback to the client in the meantime :)